src/common/ConnectedHost.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { BaseStore } from './BaseStore';
+import { BaseComponent, autobind, assign, shallowCompare } from '../Utilities';
+
+// Track all components that require changes.
+let _changedComponents: ConnectedHost[];
+
+export interface IConnectedHostProps {
+  componentProps: any;
+  storesToSubscribe: string[];
+  component: any;
+  getProps: (stores: any, props: any) => any;
+}
+
+export interface IConnectedHostState {
+  props: any;
+}
+
+export class ConnectedHost extends BaseComponent<IConnectedHostProps, IConnectedHostState> {
+  public static contextTypes = {
+    stores: React.PropTypes.object
+  };
+
+  public context: {
+    stores: {
+      [key: string]: BaseStore
+    };
+  };
+
+  private _stores: any[];
+  private _changeEnqueued: boolean;
+  private _isMounted: boolean;
+
+  constructor(props: IConnectedHostProps) {
+    super(props);
+
+    this.state = {
+      props: null
+    };
+  }
+
+  public componentWillMount() {
+    if (this.props.storesToSubscribe && this.props.storesToSubscribe.length > 0) {
+      if (!this.context.stores) {
+        throw `A connected component was hosted in an environment where no stores were hosted. Use the StoreHost to host components.`;
+      }
+
+      // Resolve and subscribe to stores.
+      this._stores = this.props.storesToSubscribe.map(storeName => {
+        let store = this.context.stores[storeName];
+
+        if (!store) {
+          throw `The "${storeName}" store was required by a connected component, but not exposed.`;
+        }
+        this._disposables.push(store.subscribe(this._onStoreChanged));
+
+        return store;
+      });
+    }
+
+    // We can only initialize state at this point, where context has been resolved.
+    this.state = {
+      props: this._getComponentProps(this.props)
+    };
+  }
+
+  public componentDidMount() {
+    this._isMounted = true;
```

Comment:
After reading the docs they're trying to discourage the use of .isMounted() method as an anti pattern, but for some reason explicitly OK certain situations where you can't simply "cancel" operations and definitely need to check. I think this is one of those cases; it is more efficient to track a boolean than it is to find and remove entries in an array.